CONTRIBUTING.md

```diff
@@ -1,28 +1,100 @@
-(even if it was for a different project), you probably don't need to do it
-again.
+## How to become a contributor and submit your own code
 
-## Code reviews
+### Contributor License Agreements
 
-All submissions, including submissions by project members, require review. We
-use GitHub pull requests for this purpose. Consult
-[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
-information on using pull requests.
+We'd love to accept your patches! Before we can take them, we have to jump a couple of legal hurdles.
 
-## Community Guidelines
+Please fill out either the individual or corporate Contributor License Agreement (CLA).
 
-This project follows [Google's Open Source Community
-Guidelines](https://opensource.google/conduct/).
+  * If you are an individual writing original source code and you're sure you own the intellectual property, then you'll need to sign an [individual CLA](https://code.google.com/legal/individual-cla-v1.0.html).
+  * If you work for a company that wants to allow you to contribute your work, then you'll need to sign a [corporate CLA](https://code.google.com/legal/corporate-cla-v1.0.html).
+
+Follow either of the two links above to access the appropriate CLA and instructions for how to sign and return it. Once we receive it, we'll be able to accept your pull requests.
+
+***NOTE***: Only original source code from you and other people that have signed the CLA can be accepted into the main repository.
+
+## Contributing code
+
+If you have improvements to the Model Card Toolkit, send us your pull requests! For those
+just getting started, Github has a [howto](https://help.github.com/articles/using-pull-requests/).
+
+### Code Ownership
+
+* Code ownership is tracked through the `CODEOWNERS` file. Users can be added if one of the following situations apply:
+  * Developers who contribute or maintain the Model Card Toolkit can gain coder owner access to the project.
+  * Requesting project code ownership requires a substantial contribution (e.g. contributed to the bi-weekly meetings, maintained some code over some time, or contributed a major feature).
+
+
+### Development tips
+
+We use [pre-commit](https://pre-commit.com/) to validate our code before we push to the repository. We use push over commit to allow more flexibility.
+
+Here's how to install it locally:
+- Create virtual environemnt: `python3 -m venv env`
+- Activate virtual environment: `source env/bin/activate`
+- Upgrade pip: `pip install --upgrade pip`
+- Install test packages: `pip install -e ".[test]"`
+- Install pre-commit hooks for push hooks: `pre-commit install --hook-type pre-push`
+- Change and commit files. pre-commit will run the tests and linting before you push. You can also manually trigger the tests and linting with `pre-commit run --hook-stage push --all-files`
+
+Note that pre-commit will be run via GitHub Action automatically for new PRs.
+
+### Contribution guidelines and standards
+
```

Comment:
Let's add a documentation section about updating docs/docstrings as part of code changes.